lang/python314-devel/files/patch-pylifecycle-compatibility.diff

```diff
@@ -0,0 +1,89 @@
+Backport of https://github.com/python/cpython/pull/128165 with minor modifications to not use legacysupport.
+--- Python/pylifecycle.c
++++ Python/pylifecycle.c
+@@ -46,8 +46,25 @@
+ 
+ #if defined(__APPLE__)
+ #  include <AvailabilityMacros.h>
++#  define TARGET_OS_OSX 1
+ #  include <mach-o/loader.h>
+-#  include <os/log.h>
++// The os_log unified logging APIs were introduced in macOS 10.12, iOS 10.0,
++// tvOS 10.0, and watchOS 3.0;
++#  if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
++#    define HAS_APPLE_SYSTEM_LOG 1
++#  elif defined(TARGET_OS_OSX) && TARGET_OS_OSX
++#    if defined(MAC_OS_X_VERSION_10_12) && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_12
++#      define HAS_APPLE_SYSTEM_LOG 1
++#    else
++#      define HAS_APPLE_SYSTEM_LOG 0
++#    endif
++#  else
++#    define HAS_APPLE_SYSTEM_LOG 0
++#  endif
```

Comment:
This is a lot more complicated than it needs to be, and upstream isn't going to accept unconditionally defining `TARGET_OS_OSX`. You know you are on macOS if it is undefined, so the correct condition would be something like:
```
#if (defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE) \
     || ((!defined(TARGET_OS_OSX) || TARGET_OS_OSX) \
         && MAC_OS_X_VERSION_MIN_REQUIRED >= 101200)
#define HAS_APPLE_SYSTEM_LOG 1
#else
#define HAS_APPLE_SYSTEM_LOG 0
#endif
```

Reply:
I don't understand the distinction, since I just replaced `#  include <TargetConditionals.h>` with `#  define TARGET_OS_OSX 1` and the upsream still doesn't have that code.